src/main/kotlin/com/parmet/buf/gradle/GenerateTask.kt

```diff
@@ -26,10 +26,43 @@ abstract class GenerateTask : DefaultTask() {
         execBuf(args + additionalArgs())
     }
 
-    private fun additionalArgs() =
-        if (getExtension().generateOptions?.includeImports == true) {
+    private fun additionalArgs(): List<String> {
+        val generateOptions = getExtension().generateOptions
+        val importOptions = if (generateOptions?.includeImports == true) {
             listOf("--include-imports")
         } else {
             emptyList()
         }
+
+        val configOptions = genConfigFile()?.let {
+            listOf("--template", it.absolutePath)
+        } ?: emptyList()
+
+        return importOptions + configOptions
+    }
+
+    private fun genConfigFile(): File? {
+        return getExtension().generateOptions?.let { generateOptions ->
+            val selectedTemplateFiles = listOfNotNull(generateOptions.templateFileLocation, project.file("buf.gen.yaml")).filter {
+                it.exists() && it.isFile
+            }
```

Comment:
Minor formatting nit - I like to respect the [rectangle rule](https://github.com/google/google-java-format/wiki/The-Rectangle-Rule) whenever possible, which would format like this:

```kotlin
val templateFiles =
    listOfNotNull(...)
        .filter { it.exists() && it.isFile }
```